### ISSUE TITLE
ReferenceError: 'globalThis' is not defined

### ISSUE THREAD

**Issue by seanemmer** [FirebaseExtended/reactfire#251]
Browser: Edge 18

The docs state to polyfill globalThis but I have had no success with the following in my application root:

require('globalthis')()
globalThis = require('globalthis')()
window.globalThis = require('globalthis')()
global.globalThis = require('globalthis')()

**Comment by jhuleatt** [comment]
@jamesdaniels are you able to help out with this?

**Comment by nirtamir2** [comment]
I had the same issue, but I fixed it
I'm not the maintainer of this library or `globalThis` polyfill library so my answer may not be what you really need (I created a react app from CRA):

I added in my `index.ts` file (the first file your app load)
```typscript
import { shim } from "globalthis";
import { loadApp } from "./loadApp";

shim();
loadApp();
```
`loadApp` is loading your app after the `shim` function that does the "polyfilling".
So if your app uses `reactfire` you will have the polyfill before it runs

**Comment by nirtamir2** [comment]
I had the same issue.

I'm using CRA.
I installed `globalthis` (because its bundle size is smaller than `core-js/features/global-this`)

I created a file `./polifills.ts` with
```typescript
// polyfills.ts
import globalthis from "globalthis";
globalthis.shim();
```

And import it as the first import in `index.tsx`.
```typescript
// index.tsx
import "./polyfills"
...
```

**NOTE:** if you try to debug your code in CRA for old browsers compatibility, make sure you `browserslist` entry in `package.json` for development match the same as production.

I tested with edge 18 and it worked.
